modules/mobile-agents/android/client/plugins/IDPProxy/src/org/wso2/mdm/agent/proxy/utils/ServerUtilities.java

```diff
@@ -0,0 +1,375 @@
+	private final static String TAG = "ServerUtilities";
+	private static final DateFormat dateFormat = new SimpleDateFormat("MM/dd/yyyy HH:mm:ss",
+	                                                                  Locale.getDefault());
+
+	/**
+	 * Validate the token expiration date.
+	 *
+	 * @param expirationDate - Token expiration date.
+	 * @return - Token status.
+	 */
+	public static boolean isValid(Date expirationDate) {
+		Date currentDate = new Date();
+		String formattedDate = dateFormat.format(currentDate);
+		currentDate = convertDate(formattedDate);
+
+		boolean isExpired = currentDate.after(expirationDate);
+		boolean isEqual = currentDate.equals(expirationDate);
+		if (isExpired == true || isEqual == true) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Convert the date to the standard format.
+	 *
+	 * @param date - Date as a string.
+	 * @return - Formatted date.
+	 */
+	public static Date convertDate(String date) {
+		Date receivedDate = null;
+		try {
+			receivedDate = dateFormat.parse(date);
+		} catch (ParseException e) {
+			Log.e(TAG, "Invalid date format." + e);
+		}
+
+		return receivedDate;
+	}
+
+	public static String buildPayload(Map<String, String> params) {
+		if (params == null) {
+			return null;
+		}
+		StringBuilder bodyBuilder = new StringBuilder();
+		Iterator<Entry<String, String>> iterator = params.entrySet().iterator();
+		while (iterator.hasNext()) {
+			Entry<String, String> param = iterator.next();
+			bodyBuilder.append(param.getKey()).append('=').append(param.getValue());
+			if (iterator.hasNext()) {
+				bodyBuilder.append('&');
+			}
+		}
+		return bodyBuilder.toString();
+	}
+
+	public static HttpRequestBase buildHeaders(HttpRequestBase httpRequestBase,
+	                                           Map<String, String> headers, String httpMethod) {
+		Iterator<Entry<String, String>> iterator = headers.entrySet().iterator();
+		while (iterator.hasNext()) {
+			Entry<String, String> header = iterator.next();
+			httpRequestBase.setHeader(header.getKey(), header.getValue());
+		}
+		return httpRequestBase;
+	}
+
+	public static Map<String, String> postData(APIUtilities apiUtilities,
+	                                           Map<String, String> headers)
+			throws IDPTokenManagerException {
+
+		String httpMethod = apiUtilities.getHttpMethod();
+		String message;
+		String url = apiUtilities.getEndPoint();
+		JSONObject params = apiUtilities.getRequestParams();
+		Map<String, String> responseParams = new HashMap<String, String>();
+		HttpClient httpClient = getCertifiedHttpClient();
+
+		if (httpMethod.equals(Constants.POST_METHOD)) {
+			HttpPost httpPost = new HttpPost(url);
+			if (params != null) {
+				try {
+					httpPost.setEntity(new StringEntity(params.toString()));
+				} catch (UnsupportedEncodingException e) {
+					message = "Invalid encoding type.";
+					throw new IDPTokenManagerException(message, e);
+				}
+			} else {
+				httpPost.setEntity(null);
```

Comment:
Do we specifically need to set the entity of the request to null? Can't we leave the defaults as it is?